src/main/java/net/javadiscord/javabot/systems/qotw/QOTWCloseSubmissionsJob.java

```diff
@@ -55,23 +61,51 @@ public void execute() throws SQLException {
 					.queue();
 			if (config.getModerationConfig().getLogChannel() == null) continue;
 			if (qotwConfig.getSubmissionChannel() == null || qotwConfig.getQuestionChannel() == null) continue;
-			Message message = getLatestQOTWMessage(qotwConfig.getQuestionChannel(), qotwConfig, jda);
-			if (message == null) continue;
-			message.editMessageComponents(ActionRow.of(Button.secondary("qotw-submission:closed", "Submissions closed").asDisabled())).queue();
-			for (ThreadChannel thread : qotwConfig.getSubmissionChannel().getThreadChannels()) {
-				try (Connection con = dbHelper.getDataSource().getConnection()) {
-					Optional<QOTWSubmission> optionalSubmission = qotwSubmissionRepository.getSubmissionByThreadId(thread.getIdLong());
-					if (optionalSubmission.isEmpty()) continue;
-					new SubmissionControlsManager(botConfig.get(guild), optionalSubmission.get(), pointsService, notificationService).sendControls();
-				} catch (SQLException e) {
-					ExceptionLogger.capture(e, getClass().getSimpleName());
-					throw e;
+			Message questionMessage = getLatestQOTWMessage(qotwConfig.getQuestionChannel(), qotwConfig, jda);
```

Comment:
What if that message doesn't exist?
In the past, it often happened that the initial QOTW message wasn't sent.

Reply:
what? where did that ever happen?
it was always like that

Reply:
This is checking the initial message of the bit right? In previous QOTWs, the bot didn't add users to their threads (missing the initial message).

Reply:
no, this is simply retrieving the qotw "announcement" message

Reply:
Ok, I just misunderstood that.